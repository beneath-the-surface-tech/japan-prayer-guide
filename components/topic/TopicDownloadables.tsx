import React from "react"
import { Container, Card } from "react-bootstrap"
import Button from "react-bootstrap/Button"
import Row from "react-bootstrap/Row"
import Col from "react-bootstrap/Col"
import Link from "next/link"
import { useTranslation } from "react-i18next"

interface downloadProps {
    links: string[]
    title: string
    labels: string[]
    headers: string[]
}

export default function TopicDownloadables({ links, title, labels, headers }: downloadProps) {
    const { i18n } = useTranslation("common")
    return (
        <Container
            data-testid={"topic-downloadables-container"}
            className="d-flex flex-column my-5"
            id="topic-downloads"
        >
            <Container className="d-flex flex-row justify-content-between align-items-center">
                <h1 data-testid={"topic-downloadables-title"} className="text-primary my-4 fs-1">
                    {title}
                </h1>
                <Link href={"/resources"} className="text-secondary d-none d-md-block" locale={i18n.language}>
                    {headers[0]}
                </Link>
            </Container>
            <Row xl={4} md={2} className="g-3" data-testid={"topic-downloadables-links"}>
                {labels.map((label, idx) => (
                    <Col key={idx + label}>
                        <Link href={links[idx]} className="text-decoration-none" locale={i18n.language}>
                            <Card
                                className="shadow-sm border-0 rounded"
                                style={{ backgroundColor: "#E2E2E2", minHeight: "125px" }}
                            >
                                <Card.Body className="d-flex align-items-center justify-content-center">
                                    <p className="text-center fw-bold fs-4 my-0">{label}</p>
                                </Card.Body>
                            </Card>
                        </Link>
                    </Col>
                ))}
            </Row>
            <Link
                href={links[4]}
                className="align-self-center w-100"
                style={{ maxWidth: "500px" }}
                locale={i18n.language}
            >
                <Button
                    className="align-self-center w-100 mt-4 text-white border-secondary bg-secondary"
                    variant="primary"
                >
                    {headers[1]}
                </Button>
            </Link>
            <Link
                href={"/resources"}
                className="align-self-center my-3 text-secondary d-md-none"
                locale={i18n.language}
            >
                {headers[0]}
            </Link>
        </Container>
    )
}
