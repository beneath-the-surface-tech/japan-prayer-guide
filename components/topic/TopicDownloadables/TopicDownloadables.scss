#topic-downloads {
    h2 {
        font-family: $font-family-secondary;
    }

    p {
        font-size: 20px;
        @include media-breakpoint-down(md) {
            font-size: 18px;
        }
    }

    .btn {
        font-size: 19px;
        @include media-breakpoint-down(md) {
            font-size: 16px;
        }
    }
}
