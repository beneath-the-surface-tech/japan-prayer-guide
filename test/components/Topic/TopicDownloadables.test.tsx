import { render, screen } from '@testing-library/react';
import TopicDownloadables from '../../../components/topic/TopicDownloadables';

describe("Topic Downloadables", () => {
    test("Renders a section with the right text inside", () => {
        const testTextArray = ["Hey how's it going", "I'm doing fine", "Thanks", "great"]
        const testTextArray2 = ["/", "/", "/", "/", "/"]
        const testTitle = "Topic Downloadables"
        render(<TopicDownloadables labels={testTextArray} links={testTextArray2} title={testTitle}/>)
        const topicDownloadablesCont = screen.getByTestId("topic-downloadables-container")
        const topicDownloadablesTitle = screen.getByTestId("topic-downloadables-title")
        const topicDownloadablesLinks = screen.getByTestId("topic-downloadables-links").children

        expect(topicDownloadablesCont).toHaveClass("d-flex", "flex-column", "my-5")

        expect(topicDownloadablesTitle).toHaveTextContent(testTitle)
        expect(topicDownloadablesTitle).toHaveClass("text-primary", "my-4", "fs-1")

        expect(topicDownloadablesLinks.length).toBe(4)
        expect(topicDownloadablesLinks[0]).toHaveTextContent(testTextArray[0])
        expect(topicDownloadablesLinks[0]).toHaveClass("col")
    })
})