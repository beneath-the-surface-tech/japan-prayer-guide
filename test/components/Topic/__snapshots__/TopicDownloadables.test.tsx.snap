// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Topic Downloadables Snapshot test 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="d-flex flex-column my-5 container"
        data-testid="topic-downloadables-container"
        id="topic-downloads"
      >
        <div
          class="d-flex flex-row justify-content-between align-items-center container"
        >
          <h1
            class="text-primary my-4 fs-1"
            data-testid="topic-downloadables-title"
          >
            Downloadables
          </h1>
          <a
            class="text-secondary d-none d-md-block"
            href="/resources"
          >
            hi
          </a>
        </div>
        <div
          class="g-3 row row-cols-xl-4 row-cols-md-2"
          data-testid="topic-downloadables-links"
        >
          <div
            class="col"
          >
            <a
              class="text-decoration-none"
              href="/"
            >
              <div
                class="shadow-sm border-0 rounded card"
                style="background-color: rgb(226, 226, 226); min-height: 125px;"
              >
                <div
                  class="d-flex align-items-center justify-content-center card-body"
                >
                  <p
                    class="text-center fw-bold fs-4 my-0"
                  >
                    ab
                  </p>
                </div>
              </div>
            </a>
          </div>
          <div
            class="col"
          >
            <a
              class="text-decoration-none"
              href="/"
            >
              <div
                class="shadow-sm border-0 rounded card"
                style="background-color: rgb(226, 226, 226); min-height: 125px;"
              >
                <div
                  class="d-flex align-items-center justify-content-center card-body"
                >
                  <p
                    class="text-center fw-bold fs-4 my-0"
                  >
                    cd
                  </p>
                </div>
              </div>
            </a>
          </div>
          <div
            class="col"
          >
            <a
              class="text-decoration-none"
              href="/"
            >
              <div
                class="shadow-sm border-0 rounded card"
                style="background-color: rgb(226, 226, 226); min-height: 125px;"
              >
                <div
                  class="d-flex align-items-center justify-content-center card-body"
                >
                  <p
                    class="text-center fw-bold fs-4 my-0"
                  >
                    ef
                  </p>
                </div>
              </div>
            </a>
          </div>
        </div>
        <a
          class="align-self-center w-100"
          href="/"
          style="max-width: 500px;"
        >
          <button
            class="align-self-center w-100 mt-4 text-white border-secondary bg-secondary btn btn-primary"
            type="button"
          >
            test
          </button>
        </a>
        <a
          class="align-self-center my-3 text-secondary d-md-none"
          href="/resources"
        >
          hi
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="d-flex flex-column my-5 container"
      data-testid="topic-downloadables-container"
      id="topic-downloads"
    >
      <div
        class="d-flex flex-row justify-content-between align-items-center container"
      >
        <h1
          class="text-primary my-4 fs-1"
          data-testid="topic-downloadables-title"
        >
          Downloadables
        </h1>
        <a
          class="text-secondary d-none d-md-block"
          href="/resources"
        >
          hi
        </a>
      </div>
      <div
        class="g-3 row row-cols-xl-4 row-cols-md-2"
        data-testid="topic-downloadables-links"
      >
        <div
          class="col"
        >
          <a
            class="text-decoration-none"
            href="/"
          >
            <div
              class="shadow-sm border-0 rounded card"
              style="background-color: rgb(226, 226, 226); min-height: 125px;"
            >
              <div
                class="d-flex align-items-center justify-content-center card-body"
              >
                <p
                  class="text-center fw-bold fs-4 my-0"
                >
                  ab
                </p>
              </div>
            </div>
          </a>
        </div>
        <div
          class="col"
        >
          <a
            class="text-decoration-none"
            href="/"
          >
            <div
              class="shadow-sm border-0 rounded card"
              style="background-color: rgb(226, 226, 226); min-height: 125px;"
            >
              <div
                class="d-flex align-items-center justify-content-center card-body"
              >
                <p
                  class="text-center fw-bold fs-4 my-0"
                >
                  cd
                </p>
              </div>
            </div>
          </a>
        </div>
        <div
          class="col"
        >
          <a
            class="text-decoration-none"
            href="/"
          >
            <div
              class="shadow-sm border-0 rounded card"
              style="background-color: rgb(226, 226, 226); min-height: 125px;"
            >
              <div
                class="d-flex align-items-center justify-content-center card-body"
              >
                <p
                  class="text-center fw-bold fs-4 my-0"
                >
                  ef
                </p>
              </div>
            </div>
          </a>
        </div>
      </div>
      <a
        class="align-self-center w-100"
        href="/"
        style="max-width: 500px;"
      >
        <button
          class="align-self-center w-100 mt-4 text-white border-secondary bg-secondary btn btn-primary"
          type="button"
        >
          test
        </button>
      </a>
      <a
        class="align-self-center my-3 text-secondary d-md-none"
        href="/resources"
      >
        hi
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
